deployment/ccip/changeset/v1_6/cs_add_new_chain_e2e.go

```diff
@@ -0,0 +1,932 @@
+package v1_6
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"math/big"
+
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/smartcontractkit/ccip-owner-contracts/pkg/proposal/timelock"
+	mcmslib "github.com/smartcontractkit/mcms"
+	mcmssdk "github.com/smartcontractkit/mcms/sdk"
+	mcmstypes "github.com/smartcontractkit/mcms/types"
+
+	"github.com/smartcontractkit/chainlink/deployment"
+	"github.com/smartcontractkit/chainlink/deployment/ccip/changeset"
+	"github.com/smartcontractkit/chainlink/deployment/ccip/changeset/globals"
+	commoncs "github.com/smartcontractkit/chainlink/deployment/common/changeset"
+	"github.com/smartcontractkit/chainlink/deployment/common/proposalutils"
+	commontypes "github.com/smartcontractkit/chainlink/deployment/common/types"
+
+	"github.com/smartcontractkit/chainlink/v2/core/capabilities/ccip/types"
+	"github.com/smartcontractkit/chainlink/v2/core/gethwrappers/ccip/generated/v1_6_0/fee_quoter"
+)
+
+/*
+Future improvements:
+- Enable connecting directly to the prod router through PromoteNewChainForTestingChangeset
+- Align configuration structs with whatever is simplest for BIX team to use
+- Add more validation coverage to the precondition functions
+*/
+
+var (
+	// AddCandidatesForNewChainChangeset deploys a new chain and adds its exec and commit plugins as candidates on the home chain.
+	AddCandidatesForNewChainChangeset = deployment.CreateChangeSet(addCandidatesForNewChainLogic, addCandidatesForNewChainPrecondition)
+	// PromoteNewChainForTestingChangeset promotes exec and commit plugin candidates for the new chain on the home chain.
+	// It also connects the new chain to various destination chains through the test router.
+	// This changeset should be run after AddCandidatesForNewChainChangeset.
+	PromoteNewChainForTestingChangeset = deployment.CreateChangeSet(promoteNewChainForTestingLogic, promoteNewChainForTestingPrecondition)
+	// ConnectNewChainChangeset activates connects a new chain with other chains by updating onRamp, offRamp, and router contracts.
+	// If connecting to production routers, you should have already run PromoteNewChainForTestingChangeset.
+	ConnectNewChainChangeset = deployment.CreateChangeSet(connectNewChainLogic, connectNewChainPrecondition)
```

Comment:
Let's mention 
- if any of these changesets can be run multiple times to expect same results
- The behavior for timelock owned contracts

Reply:
https://github.com/smartcontractkit/chainlink/pull/16705/commits/8104b00de8d2820f96e1d73a532a8a32c98f51f0